src/AsimovUtils.cc

```diff
@@ -21,8 +21,7 @@ RooAbsData *asimovutils::asimovDatasetNominal(RooStats::ModelConfig *mc, double
 
 	if (verbose>2) {
 	    Logger::instance().log(std::string(Form("AsimovUtils.cc: %d -- Parameters after fit for asimov dataset",__LINE__)),Logger::kLogLevelInfo,__func__);
-    	    std::unique_ptr<TIterator> iter(mc->GetPdf()->getParameters((const RooArgSet*) 0)->createIterator());
-    	    for (RooAbsArg *a = (RooAbsArg *) iter->Next(); a != 0; a = (RooAbsArg *) iter->Next()) {
+    	    for (RooAbsArg *a : *std::unique_ptr<RooArgSet>{mc->GetPdf()->getParameters((const RooArgSet*) 0)}) {
```

Comment:
Not that a memory leak is fixed  here by wrapping the return value of `getParameters` in a smart pointer (`getParameters` returns a pointer that needs to be deleted)

Reply:
This change has introduced a seg fault when running with -v 3, try for example 

`combine data/tutorials/CAT23001/parametric-analysis-datacard.txt -m 125 -v 3`

Any thoughts @guitargeek ?

Reply:
It could be that some compilers don't like that this `std::unique_ptr<RooArgSet>` objects lifetime is only defined by the scope of the loop.

What happens if you give this a name?
```c++
std::unique_ptr<RooArgSet> paramsTmp{mc->GetPdf()->getParameters((const RooArgSet*) 0)};
for (RooAbsArg *a : *paramsTmp) {
}
```

Reply:
Alternatively you can try the `getParameters` overload with an output parameter, which doesn't do a heap allocation so in principle it's even better:
```c++
RooArgSet paramsTmp;
mc->GetPdf()->getParameters(nullptr, paramsTmp);
for (RooAbsArg *a : *paramsTmp) {
}
```

Let me know if these patterns avoid the segfault